### PR TITLE
Additional unit tests for datastore access methods

### PR DESCRIPTION
**Pull request by tatiana-s**
Main change: some additional unit tests for most methods in `DatastoreAccess` as well as some changes to the existing ones.

Also integrated the Jacoco Plugin to get test coverage. To see the coverage report, run `mvn verify`, then navigate to `target/site/jacoco/index.html` and open as window. For example, the current report (76% line coverage in general):

Screenshot 2020-09-08 at 11 07 28 PM

Tests related to messages will be added once issue #22 is implemented, and some tests for joining events are also still missing (edit: added test for joining events, raising coverage to 87%).

As an additional bug fix, `addEventToGroup` now uses a cross-group transaction instead of multiple transaction to avoid events being added to the database without being associated with a group if one of the transactions fails.

### CODE REVIEW

File: src/main/java/com/google/lecturechat/data/DatastoreAccess.java [hunk 1]
```diff
@@ -176,8 +177,10 @@ public boolean isUserRegistered(String userId) {
    */
   public long addEventToGroup(
       long groupId, String title, long startTime, long endTime, String creator) {
-    Transaction eventTransaction = datastore.beginTransaction();
+    TransactionOptions options = TransactionOptions.Builder.withXG(true);
```
Comment (csperl) [inline src/main/java/com/google/lecturechat/data/DatastoreAccess.java:180]: Nit: Inline this variable.
Also, add a one line comment here, saying that you create a cross-group transaction to make both operations atomic.

File: src/main/java/com/google/lecturechat/data/DatastoreAccess.java [hunk 2]
```diff
@@ -221,7 +218,7 @@ public long addEventToGroup(
    * @param groupId The id of the group.
    * @return The list of events.
    */
-  private List<Event> getAllEventsFromGroup(long groupId) {
+  public List<Event> getAllEventsFromGroup(long groupId) {
```
Comment (csperl) [inline src/main/java/com/google/lecturechat/data/DatastoreAccess.java:221]: This does not need to be public. Now you are using it in tests, therefore needs to be at least package-private.

File: src/test/java/com/google/lecturechat/DatastoreAccessTest.java [hunk 3]
```diff
@@ -31,12 +31,29 @@
...
   private final LocalServiceTestHelper helper =
       new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
   private DatastoreAccess datastore;
   private DatastoreService service;
 
+  // Parameters for arranging.
+  private final String UNIVERSITY_A = "A";
+  private final String UNIVERSITY_B = "B";
+  private final String UNIVERSITY_C = "C";
+  private final String DEGREE = "A";
```
Comment (csperl) [inline src/test/java/com/google/lecturechat/DatastoreAccessTest.java:43]: Nit, use other symbols here and not always "A". If your code under test suddenly returns a university or a degree but you expect the event title, then you would still match.

**Review by csperl — APPROVED** [review]
Really cool! I particularily like the overview of the test coverage :)

**Review by AlexandraImbrisca — APPROVED** [review]

**Pull request merged**